src/scheduler.cpp

```diff
@@ -16,22 +15,14 @@ CScheduler::CScheduler() : nThreadsServicingQueue(0), stopRequested(false), stop
 
 CScheduler::~CScheduler()
 {
+    LOCK(newTaskMutex);
```

Comment:
Adding a lock in a destructor just for an assertion is a shame. Maybe this should be behind a debug guard?

Reply:
It's probably not useful in the first place, the lock's about to be destroyed, so if anyone else even has a reference to it there's a bug, and threadsafety annotations don't apply to constructors and destructors for the same reason.